sdks/wasm/Mono.WebAssembly.DebuggerProxy/FirefoxMonoProxy.cs

```diff
@@ -0,0 +1,496 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Net.Sockets;
+using System.Net;
+using System.Text;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace WsProxy {
+	public class FirefoxMonoProxyServer 
+	{
+		int portProxy;
+		int portBrowser;
+
+		public FirefoxMonoProxyServer (int portProxy, int portBrowser)
+		{
+			this.portBrowser = portBrowser;
+			this.portProxy = portProxy;
+		}
+
+		public void Run ()
+		{
+			var _server = new TcpListener (IPAddress.Parse ("127.0.0.1"), portProxy);
+			_server.Start ();
+			while (true) {
+				// wait for client connection
+				TcpClient newClient = _server.AcceptTcpClient ();
+
+				// client found.
+				// create a thread to handle communication
+				var monoProxy = new FirefoxMonoProxy (portBrowser);
+				Thread t = new Thread (new ParameterizedThreadStart (monoProxy.Run));
+				t.Start (newClient);
+			}
+		}
+
+			
+	}
+	public class FirefoxMonoProxy : MonoProxy {
+		string actorName;
+		string readedIDE;
+		string readedBrowser;
+		int local_breakpoint_id;
+		bool isPausedOnMonoBreakpoint;
+		int portBrowser;
+
+		public FirefoxMonoProxy (int portBrowser)
+		{
+			this.portBrowser = portBrowser;
+		}
+		public void Run (object obj)
+		{
+			TcpClient client = (TcpClient)obj;
+			NetworkStream ide = client.GetStream ();
+
+			TcpClient client2 = new TcpClient ();
+			client2.Connect ("127.0.0.1", portBrowser);
+			NetworkStream Browser = client2.GetStream ();
+
+			while (true) {
```

Comment:
In follow-up work it would be nice to change this look to use async reads so that you don't have this busy loop.